### PR TITLE
transfer: fix workersrcs argument parsing in MultipleFileUpload

### LINKED ISSUE #3167: Giving a Property as workersrcs to MultipleFileUpload breaks
This issue was introduced by 4173daceb97c6e2583e7be9bc63ed4dcf7ba8f2e.

I give a Property parameter to MultipleFileUploads, the property resolves to a list.

Due to the mentioned commit, The value is now a list of list, and this breaks.

``` pytb
2017-05-02 12:55:50+0200 [-] MultipleFileUpload started, from worker [[u'build/filename']] to master '/home/buildsystem/master/public_html/destination'
2017-05-02 12:55:50+0200 [-] BuildStep.failed; traceback follows
        Traceback (most recent call last):
          File "/home/buildsystem/bb-master/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1357, in gotResult
            _inlineCallbacks(r, g, deferred)
          File "/home/buildsystem/bb-master/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1301, in _inlineCallbacks
            result = g.send(result)
          File "/home/buildsystem/bb-master/local/lib/python2.7/site-packages/buildbot/process/buildstep.py", line 547, in startStep
            self.results = yield self.run()
          File "/home/buildsystem/bb-master/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1447, in unwindGenerator
            return _inlineCallbacks(None, gen, Deferred())
        --- <exception caught here> ---
          File "/home/buildsystem/bb-master/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1301, in _inlineCallbacks
            result = g.send(result)
          File "/home/buildsystem/bb-master/local/lib/python2.7/site-packages/buildbot/process/buildstep.py", line 677, in run
            results = yield self._start_deferred
        exceptions.AttributeError: 'list' object has no attribute 'rfind'
```

### PR DESCRIPTION
**Pull request by noc0lour**
## Contributor Checklist:

* [ ] I have updated the unit tests
* [ ] I have created a file in the master/buildbot/newsfragment directory (and read the README.txt in that directory)
* [ ] I have updated the appropriate documentation

This should fix #3167.
@benallard Can you add a unit test which represents your usecase/missing corner cases?

### CODE REVIEW

**Comment by tardyp** [comment]
As pointed out by @benallard, Please add a documentation snippet explaining that multifile upload accept both list or string for workersrcs

**Comment by noc0lour** [comment]
If we only want to support lists as argument I'm fine with that as well, then we would have to throw an error after checking if it is a list.

**Comment by tardyp** [comment]
I think having a single string makes sense for glob, and the code is simple enough and tested. We just need it to be documented

**Comment by noc0lour** [comment]
Sure, I'll look at it later today ;)

**Comment by noc0lour** [comment]
Added a sentence to the docs (:

File: master/docs/manual/cfg-buildsteps.rst [hunk 1]
```diff
@@ -2204,7 +2204,7 @@ Additionally it supports the ``glob`` parameter if this parameter is set to ``Tr
     f.addStep(steps.MultipleFileUpload(workersrcs=["docs", "test-results.html"],
                                        masterdest="/usr/share/nginx/www/",
                                        url="~buildbot"))
-
+..
```
Comment (tardyp) [inline master/docs/manual/cfg-buildsteps.rst]: looks like those two dots are not wanted

Comment (noc0lour) [inline master/docs/manual/cfg-buildsteps.rst]: Yep, now (:
